### DISCUSSION TITLE
Using ESP-NOW in Arduino · Developer Portal

### DISCUSSION THREAD

**Discussion by giscus** [espressif/developer-portal#158] [General]
# Using ESP-NOW in Arduino · Developer Portal

The developer resources in just one place!

https://developer.espressif.com/blog/arduino-esp-now-lib/

**Comment by rajtan** [comment]
Great to see a descriptive write up of using esp-now on arduino. However, I would request and it seems very much in context that you may please provide addition example / code for using ESP-NOW LR mode in Arduino environment for ESP32 and eligible chips/modules. This would perhaps involve using the library function from ESP-IDF for setting up appropriate protocol and data rate. If there is any official example of that in Espressif's repository, please point to that. I did search but did not find ESP-NOW with LR mode example in the giit repository of examples.

**Reply by lucasssvaz** [reply]
Hi @rajtan. To use the long range mode you just need to call `WiFi.enableLongRange(true)` before setting the WiFi mode.

**Comment by f1netune** [comment]
If this protocol is to be used by collecting data from sensors, how many slave devices are supported by this protocol? Is that protocol good for gathering the sensor data? (in its determined range)

**Reply by lucasssvaz** [reply]
Hi @f1netune,

Nothing prevents you from using ESP-NOW to read sensor data. I will mostly depend on your scenario - if you need encrypted data, the size of the data, how many sensors...

The limitations of ESP-NOW are 20 total paired peers and, out of these, the maximum number of encrypted peers is limited by CONFIG_ESP_WIFI_ESPNOW_MAX_ENCRYPT_NUM (range 0–17, default 7).

**Comment by MikeyMoMo** [comment]
On the recommendation herewith, I tried adding WiFi.enableLongRange(true) as mentioned.  It made me lose communication for my transmitter that was previously working at the same spot.  Your results may vary but if you turn that on and lose signal, remember you have been warned.  When I commented it out (with a note to not use it), I started working again.  That was a test spot and the actual landing place for the sensor and ESP may be farther so I was getting ready for any move farther away from the WAP but that will be with the original code, normal range.

**Comment by NigelGuyLeeming** [comment]
Can you add an example for two intercommunicating devices without using serialnow? The simplest version please. I used ESPNow a year ago for another project and I don't recall struggling too hard. I can't get the newer version working at all.

**Reply by NigelGuyLeeming** [reply]
Not broadcast. Peer to peer.

**Reply by NigelGuyLeeming** [reply]
Looks like it wasn't a software issue after all. I was sending from a C3-supermini to an S3 T-Display. There is something not quite right with the C3 module - the proximity and orientation of the antenna to the crystal - which allows the C3 to receive but not send. The fix for the C3 was this:
  WiFi.mode(WIFI_STA);
  esp_wifi_set_max_tx_power(40);   // 0 dBm

Other suggestions are to replace the red antenna with a bit of wire, or desolder the antenna and rotate it 90 degrees away from the crystal as only one end is electrically connected anyway. Or by a better module. Mine was from AliExpress.
